app/assets/javascripts/controllers/deployment/deploy-provider-controller.js

```diff
@@ -0,0 +1,159 @@
+        masterCount: 0,
+        nodeCount: 0,
+        infraNodeCount: 0,
+        cdnConfigType: 'satellite',
+        authentication: {
+          mode: 'all'
+        }
+      };
+
+      $timeout(function() {
+        $scope.data.providers = [
+          {
+            id: 1,
+            name: 'Existing Provider 1'
+          },
+          {
+            id: 2,
+            name: 'Existing Provider 2'
+          },
+          {
+            id: 3,
+            name: 'Existing Provider 3'
+          },
+          {
+            id: 4,
+            name: 'Existing Provider 4'
+          }
+        ];
+
+        $scope.data.nodeCreationTemplates = [
+          {
+            id: 1,
+            name: 'Existing template 1'
+          },
+          {
+            id: 2,
+            name: 'Existing template 2'
+          },
+          {
+            id: 3,
+            name: 'Existing template 3'
+          },
+          {
+            id: 4,
+            name: 'Existing template 4'
+          }
+        ];
+
+        $scope.deployProviderReady = true;
+      }, 5000);
+
+      $scope.deploymentDetailsGeneralComplete = false;
+      $scope.deployComplete = false;
+      $scope.deployInProgress = false;
+      $scope.nextButtonTitle = "Next >";
+    };
+
+    $scope.ready = false;
+    var url = '/api/container_deployments';
+    $http.post(url, {"action" : "collect_data", "resource" : "none"}).success(function (response) {
+      'use strict';
+
+      $scope.deploymentData = response.results[0].data;
+      initializeDeploymentWizard();
+      $scope.ready = true;
+    });
+
+    $scope.data = {};
+    $scope.deployComplete = false;
+    $scope.deployInProgress = false;
+
+    var startDeploy = function () {
+      $scope.deployInProgress = true;
+      $timeout(function () {
+        var url = '/api/container_deployments';
+        $http.post(url, {"action" : "start", "resource" :  $scope.data}).success(function (response) {
+          'use strict';
+          $scope.deployInProgress = true;
+        });
+
+        if ($scope.deployInProgress) {
+          $scope.deployInProgress = false;
+          $scope.deployComplete = true;
+
+          $scope.deployComplete = true;
+          $scope.deploySuccess = true;
+          $scope.deployFailed = false;
+          $scope.deployFailureMessage = "An unknown error has occurred.";
+        }
+      }, 5000);
+    };
+
+    $scope.nextCallback = function(step) {
+      if (step.stepId === 'review-summary') {
+        startDeploy();
+      }
+      return true;
+    };
+    $scope.backCallback = function(step) {
+      return true;
+    };
+
+    $scope.$on("wizard:stepChanged", function(e, parameters) {
+      if (parameters.step.stepId == 'review-summary') {
+        $scope.nextButtonTitle = "Deploy";
+      } else if (parameters.step.stepId == 'review-progress') {
+        $scope.nextButtonTitle = "Close";
+      } else {
+        $scope.nextButtonTitle = "Next >";
+      }
+    });
```

Comment:
> Might be nicer to have "Next >" and the like as constants in another file

@zeari thing is, ```$scope.nextButtonTitle``` isn't a constant. In fact, as you can see from the above code, it changes from one step to another.